data_processing/cohortManager/cohortManager.py

```diff
@@ -0,0 +1,183 @@
+from checksumdir import dirhash
+
+import os, shutil, sys, importlib, json, yaml, subprocess, time, uuid, requests
+import pandas as pd
+import itk
+
+from radiomics import featureextractor  # This module is used for interaction with pyradiomics
+import threading
+
+app    = Flask(__name__)
+logger = init_logger("flask-mind-server.log")
+cfg    = ConfigSet(name=const.APP_CFG,  config_file="config.yaml")
+spark  = SparkConfig().spark_session(const.APP_CFG, "data_processing.radiology.api.5004")
+conn   = Neo4jConnection(uri=os.environ["GRAPH_URI"], user="neo4j", pwd="password")
+lock   = threading.Lock()
+
+app.config['MAX_CONTENT_LENGTH'] = 100 * 1024 * 1024
+
+STREAMS = {}
+METHODS = {}
+HOST = os.environ['HOSTNAME']
+
+# ==================================================================================================
+# Routes to list things
+# ==================================================================================================
+
+# List all the patients
+@app.route('/mind/api/v1/listPatients/<cohort_id>', methods=['GET'])
+def listPatients(cohort_id):
+    res = conn.query(f"""MATCH (px:patient) where px.cohort='{cohort_id}' and px.active=True RETURN px""")
+    all_px = []
+    for rec in res:
+        px_dict = rec.data()['px']
+        patient_id = px_dict['name']
+        px_dict['cases'] = requests.get(f'http://{HOST}:5004/mind/api/v1/listCases/{cohort_id}/{patient_id}').json()
+        all_px.append(px_dict)
+    return jsonify(all_px)
+
+# List all the cases
+@app.route('/mind/api/v1/listCases/<cohort_id>/<patient_id>', methods=['GET'])
+def listCases(cohort_id, patient_id):
+    res = conn.query(f"""MATCH (px:patient)-[:PROXY]-(proxy)-[:HAS_CASE]-(case:case) where px.name='{patient_id}' and px.cohort='{cohort_id}' and case.cohort='{cohort_id}' RETURN case""")
+    all_case = []
+    for rec in res:
+        case_dict = rec.data()['case']
+        case_id   = case_dict['AccessionNumber']
+        case_dict['scans'] = requests.get(f'http://{HOST}:5004/mind/api/v1/listScans/{cohort_id}/{patient_id}/{case_id}').json()
+        all_case.append(case_dict)
+    return jsonify(all_case)
+
+# List all the scans
+@app.route('/mind/api/v1/listScans/<cohort_id>/<patient_id>/<case_id>', methods=['GET'])
+def listScans(cohort_id, patient_id, case_id):
+    res = conn.query(f"""MATCH (px:patient)-[:PROXY]-(proxy)-[:HAS_CASE]-(case:case)-[:HAS_SCAN]-(scan:scan) where px.name='{patient_id}' and px.cohort='{cohort_id}' and case.cohort='{cohort_id}' and case.AccessionNumber='{case_id}' RETURN scan, id(scan)""")
+    all_scan = []
+    for rec in res:
+        scan_dict = rec.data()['scan']
+        scan_id   = rec.data()['id(scan)']
+        scan_dict['id']   = scan_id
+        scan_dict['data'] = requests.get(f'http://{HOST}:5004/mind/api/v1/listData/{scan_id}').json()
+        all_scan.append(scan_dict)
+    return jsonify(all_scan)
+
+# List all the scans
+@app.route('/mind/api/v1/listData/<id>', methods=['GET'])
+def listData(id):
```

Comment:
it'd be good to make the type of id more specific in the variable name and/or in the comment.